src/modules/tensor/hip/kernel/channel_dropout.cpp

```diff
@@ -0,0 +1,356 @@
+/*
+MIT License
+
+Copyright (c) 2019 - 2025 Advanced Micro Devices, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "hip_tensor_executors.hpp"
+#include <random>
+
+// -------------------- Set 0 - Dropout main kernels --------------------
+
+template<typename T>
+__device__ __forceinline__ void compute_dropout_f8(d_float8 &pix_f8, uint8_t *maskTensor, T *srcPtr)
+{
+    // Convert mask value (0 or 1) to float
+    float4 mask4 = MAKE_FLOAT4(static_cast<float>(*maskTensor));
+
+    // Multiply pixel data by mask
+    pix_f8.f4[0] = pix_f8.f4[0] * mask4;
+    pix_f8.f4[1] = pix_f8.f4[1] * mask4;
+}
+
+template<typename T>
+__device__ __forceinline__ void compute_dropout_f24(d_float24 &pix_f24, uint8_t *maskTensor, T *srcPtr)
+{
+    // Convert mask values (0 or 1) for R, G, B
+    float4 maskR_f4 = MAKE_FLOAT4(static_cast<float>(maskTensor[0]));
+    float4 maskG_f4 = MAKE_FLOAT4(static_cast<float>(maskTensor[1]));
+    float4 maskB_f4 = MAKE_FLOAT4(static_cast<float>(maskTensor[2]));
```

Comment:
The above has maskTensor as a pointer to global memory.
3 accesses to global could be written as 1 vectortype access instead?
Please check performance on this one though.

```
uchar4 mask_uc4 = *(uchar4 *) maskTensor;
float4 maskR_f4 = MAKE_FLOAT4(static_cast<float>(mask_uc4.x);
float4 maskG_f4 = MAKE_FLOAT4(static_cast<float>(mask_uc4.y);
float4 maskB_f4 = MAKE_FLOAT4(static_cast<float>(mask_uc4.z);
```